docs/README-wayland.md

```diff
@@ -59,15 +59,171 @@ successfully created, the `wl_display` and `wl_surface` objects can then be retr
-value, as SDL will handle this internally. Calls to `SDL_SetWindowSize()` should use the logical size of the window, and
-`SDL_GetWindowSizeInPixels()` should be used to query the size of the backbuffer surface in pixels. If this property is
-not set or is `SDL_FALSE`, applications can attach their own viewports or change the surface scale manually, and the SDL
-backend will not interfere or change any values internally. In this case, calls to `SDL_SetWindowSize()` should pass the
-requested surface size in pixels, not the logical window size, as no scaling calculations will be done internally.
+Custom surfaces will automatically handle scaling internally if the window was created with the
+`SDL_PROPERTY_WINDOW_CREATE_HIGH_PIXEL_DENSITY_BOOLEAN` property set to `SDL_TRUE`. In this case, applications should
+not manually attach viewports or change the surface scale value, as SDL will handle this internally. Calls
+to `SDL_SetWindowSize()` should use the logical size of the window, and `SDL_GetWindowSizeInPixels()` should be used to
+query the size of the backbuffer surface in pixels. If this property is not set or is `SDL_FALSE`, applications can
+attach their own viewports or change the surface scale manually, and the SDL backend will not interfere or change any
+values internally. In this case, calls to `SDL_SetWindowSize()` should pass the requested surface size in pixels, not
+the logical window size, as no scaling calculations will be done internally.
 
 All window functions that control window state aside from `SDL_SetWindowSize()` are no-ops with custom surfaces.
 
-Please see the minimal example in tests/testwaylandcustom.c for an example of how to use a custom, roleless surface and
-attach it to an application-managed toplevel window.
+Please see the minimal example in `tests/testwaylandcustom.c` for an example of how to use a custom, roleless surface
+and attach it to an application-managed toplevel window.
+
+## Importing external surfaces into SDL windows
+
+Wayland windows and surfaces are more intrinsically tied to the client library than other windowing systems, therefore,
+when importing surfaces, it is necessary for both SDL and the application or toolkit to use the same `wl_display`
+object. This can be set/queried via the global `SDL_PROPERTY_GLOBAL_VIDEO_WAYLAND_WL_DISPLAY_POINTER` property. To
+import an external `wl_display`, set this property before initializing the SDL video subsystem, and read the value to
+export the internal `wl_display` after the video subsystem has been initialized. Setting this property after the video
+subsystem has been initialized has no effect, and reading it when the video subsystem is uninitialized will either
+return the user provided value, if one was set prior to initialization, or NULL.
+
+This can either be obtained from the external toolkit and passed to SDL via the `SDL_ImportWaylandDisplay()`
```

Comment:
SDL_ImportWaylandDisplay doesn't exist, right?

Reply:
Yes, that paragraph should have been deleted. Fixed.